# TP 10 - Preguntas y respuestas sobre el dataset de felinos
link del dataset:https://archivos.linti.unlp.edu.ar/index.php/s/5iC9FrYWiGi11d0

In [2]:
import pandas as pd
import plotly.express as px

felinos = pd.read_csv("felinos_filtrado.csv")
print(felinos.shape)
felinos.head()


(2019, 19)


,Unnamed: 0,kingdom,phylum,class,order,family,genus,species,locality,stateProvince,lat,lng,elevation,day,month,year,taxonKey,ScientificName,VernacularNames
0,0,Animalia,Chordata,Mammalia,Carnivora,Felidae,Puma,Puma yagouaroundi,<p>Camino acceso al Destacamento Estero Poi</p>,Formosa,-25.12069,-58.18679,0.0,19.0,9.0,2006.0,2435146,Puma yagouaroundi,Jaguarondi Jaguarundi Yaguarundi
1,1,Animalia,Chordata,Mammalia,Carnivora,Felidae,Puma,Puma yagouaroundi,<p>Destacamento de Laguna Blanca</p>,Formosa,-25.17201,-58.13073,0.0,20.0,9.0,2006.0,2435146,Puma yagouaroundi,Jaguarondi Jaguarundi Yaguarundi
2,2,Animalia,Chordata,Mammalia,Carnivora,Felidae,Puma,Puma yagouaroundi,<p>P.N. El Palmar.</p>,Entre Ríos,-31.88614,-58.30839,24.0,8.0,9.0,2006.0,2435146,Puma yagouaroundi,Jaguarondi Jaguarundi Yaguarundi
3,3,Animalia,Chordata,Mammalia,Carnivora,Felidae,Puma,Puma yagouaroundi,"<p> RP Nº 86, Potrero Nº 15</p>",Corrientes,-28.04421,-58.12875,69.0,29.0,11.0,2010.0,2435146,Puma yagouaroundi,Jaguarondi Jaguarundi Yaguarundi
4,4,Animalia,Chordata,Mammalia,Carnivora,Felidae,Puma,Puma yagouaroundi,"<p> RP Nº 86, Potrero 2 Grande.</p>",Corrientes,-28.01889,-58.05301,74.0,17.0,6.0,2010.0,2435146,Puma yagouaroundi,Jaguarondi Jaguarundi Yaguarundi


## 1 Cual es la provincia con la mayor cantidad de avistamientos o registros de felinos documentados?

In [3]:
felinos["stateProvince"].value_counts()

stateProvince
Misiones               1253
Santa Cruz              313
Corrientes              178
Formosa                  74
Neuquén                  66
Neuquén / Río Negro      44
Chubut                   24
Santa Fe                 24
Entre Ríos               11
La Pampa                 11
Chaco                     5
Córdoba                   3
Tucumán                   1
Neuquen                   1
San Luis                  1
Jujuy                     1
Santa cruz                1
Name: count, dtype: int64

rt: la provincia con mas registros es Misiones, con 1253 avistamientos

## 2)¿Cuál es el avistamiento ubicado a mayor altitud (elevation) y a qué especie pertenece?

In [4]:
elevacion_maxima = felinos["elevation"].max()
print(elevacion_maxima)

felinos[felinos["elevation"] == elevacion_maxima]

1654.0


,Unnamed: 0,kingdom,phylum,class,order,family,genus,species,locality,stateProvince,lat,lng,elevation,day,month,year,taxonKey,ScientificName,VernacularNames
2018,2018,Animalia,Chordata,Mammalia,Carnivora,Felidae,Puma,Puma concolor,"<p>Cerro W. 47º 43,762' S y 72º 24,453' W . 22...",Santa cruz,-47.74994,-72.36557,1654.0,3.0,2.0,2008.0,2435099,Puma concolor,Red tiger Puma Cougar León colorado Onza berme...


rt: el registro mas alto esta a 1654 metros y es de Puma concolor

## 3) ¿Existe alguna especie que se encuentre distribuida en la mayoría de las provincias o su presencia está restringida a regiones específicas?

In [5]:
distribucion = felinos.groupby("species")["stateProvince"].nunique().sort_values(ascending=False)
distribucion

species
Puma concolor          15
Puma yagouaroundi       8
Leopardus geoffroyi     4
Leopardus pardalis      4
Leopardus guigna        3
Leopardus pajonales     3
Leopardus tigrinus      1
Panthera onca           1
Leopardus wiedii        1
Name: stateProvince, dtype: int64

rt: Puma concolor es la que mas se distribuye, aparece en 15 provincias distintas. las demas especies estan mas restringidas a ciertas zonas

## 4) ¿Cuántas especies distintas de la familia Felidae están representadas en este conjunto de datos?

In [6]:
especies = felinos["species"].unique()
print(len(especies))
especies

9


<StringArray>
[  'Puma yagouaroundi',       'Puma concolor',       'Panthera onca',
 'Leopardus geoffroyi',    'Leopardus guigna', 'Leopardus pajonales',
  'Leopardus tigrinus',  'Leopardus pardalis',    'Leopardus wiedii']
Length: 9, dtype: str

rt: hay 9 especies distintas de felidos en el dataset

## 5) ¿Cuál es la especie (species) o género (genus) que registra más observaciones?

In [7]:
felinos["species"].value_counts()

species
Puma concolor          1297
Puma yagouaroundi       332
Panthera onca           172
Leopardus pardalis       81
Leopardus geoffroyi      70
Leopardus guigna         32
Leopardus tigrinus       19
Leopardus wiedii          9
Leopardus pajonales       7
Name: count, dtype: int64

In [8]:
felinos["genus"].value_counts()

genus
Puma         1629
Leopardus     218
Panthera      172
Name: count, dtype: int64

rt: la especie con mas observaciones es Puma concolor con 1297 registros, y el genero es Puma con 1629

## 6) ¿En qué rango de años (year) se realizaron estas observaciones y cuál fue el año con más avistamientos registrados?

In [9]:
print("min:", felinos["year"].min())
print("max:", felinos["year"].max())

min: 1937.0
max: 2012.0


rt: van desde 1937 hasta 2012

## 7) ¿En qué año hubo más registros? Calculá el total de avistamientos por año y generá un gráfico de líneas o barras.

In [10]:
felinos_año = felinos["year"].value_counts().sort_index()
felinos_año

year
1937.0      2
1950.0      4
1951.0      1
1966.0      1
1967.0      1
1973.0      1
1974.0      1
1981.0      2
1982.0      1
1983.0      3
1984.0      3
1985.0      1
1986.0      6
1987.0     23
1988.0     13
1989.0      5
1990.0     10
1991.0      6
1992.0      6
1993.0     15
1994.0      3
1995.0     36
1996.0     18
1997.0     58
1998.0     66
1999.0     39
2000.0     47
2001.0     63
2002.0     56
2003.0    186
2004.0    227
2005.0    196
2006.0    202
2007.0    134
2008.0    211
2009.0    180
2010.0    128
2011.0     23
2012.0     14
Name: count, dtype: int64

In [11]:
px.bar(data_frame=felinos_año, x=felinos_año.index, y=felinos_año.values, title="Cantidad de avistamientos por año").show()

rt: el año con mas avistamientos fue 2004, con 227

## 8) ¿Cuáles son los meses con mayor presencia de felinos? Creá un gráfico de barras por mes para analizar la estacionalidad.

In [12]:
felinos_mes = felinos["month"].value_counts().sort_index()
felinos_mes

month
1.0     366
2.0     133
3.0      96
4.0     140
5.0     153
6.0     174
7.0     233
8.0     188
9.0     115
10.0    150
11.0    117
12.0    127
Name: count, dtype: int64

In [13]:
px.bar(data_frame=felinos_mes, x=felinos_mes.index, y=felinos_mes.values, title="Cantidad de registros por mes").show()

rt: enero (mes 1) es el que mas registros tiene, 366 en total

## 9) ¿Dónde se ubican los felinos en el país? Generá un mapa de dispersión usando la latitud y la longitud de cada registro.

In [14]:
px.scatter(data_frame=felinos, x="lng", y="lat", color="species", title="Distribucion geografica de los registros").show()

rt: se puede ver como los registros se concentran mas que nada en el noreste (Misiones) y despues hay otro grupo importante en el sur, en Santa Cruz

## 10) ¿Qué porcentaje de localidades contiene etiquetas HTML (como <p>)? Calculá la cantidad exacta de filas que necesitan limpieza.

In [15]:
con_html = felinos["locality"].fillna("").astype(str).str.contains("<p", case=False, regex=True)
cantidad_html = con_html.sum()
porcentaje_html = cantidad_html / len(felinos) * 100

print(cantidad_html)
print(round(porcentaje_html, 2))

1789
88.61


rt: hay 1789 filas que tienen etiquetas html en locality, mas o menos el 88.61% del total. estas habria que limpiarlas antes de usar el campo para otra cosa

## Conclusion

Misiones es la provincia con mas registros y tambien donde mas se concentran los avistamientos en el mapa. Puma concolor es la especie mas avistada y la mas distribuida geograficamente. La mayoria de los registros son de los años 2000, con un pico en 2004. Por ultimo, buena parte del campo locality tiene tags de html que habria que limpiar.

In [16]:
! pip install pandas


[notice] A new release of pip is available: 26.1.2 -> 26.2.1
[notice] To update, run: python.exe -m pip install --upgrade pip


In [17]:
! pip install plotly


[notice] A new release of pip is available: 26.1.2 -> 26.2.1
[notice] To update, run: python.exe -m pip install --upgrade pip


In [18]:
pip install folium

Note: you may need to restart the kernel to use updated packages.



[notice] A new release of pip is available: 26.1.2 -> 26.2.1
[notice] To update, run: python.exe -m pip install --upgrade pip


In [19]:
! pip install nbformat


[notice] A new release of pip is available: 26.1.2 -> 26.2.1
[notice] To update, run: python.exe -m pip install --upgrade pip
